src/api/categories.ts

```diff
@@ -1,18 +1,25 @@
+import { cookies } from 'next/headers';
+
 import type { CategoryType } from '@app/types/list.types';
 
 export async function getCategories(): Promise<
   { categories: CategoryType[] } | undefined
 > {
   try {
-    const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}categories`, {
-      method: 'GET',
-      headers: {
-        Authorization: `Bearer ${process.env.NEXT_PUBLIC_TOKEN}`,
-        'Content-Type': 'application/json',
-      },
-      cache: 'no-store',
-    });
-    return res.json();
+    const token = cookies().get('token');
+    if (token && token.value) {
+      const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}categories`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${token.value}`,
+          'Content-Type': 'application/json',
+        },
+        cache: 'no-store',
+      });
```

Comment:
This is an example how to make each call with a token. I added to this call, now you can add to all other calls